libkernelflinger/tpm2_security.c

```diff
@@ -941,6 +941,97 @@ static EFI_STATUS tpm2_check_bootloader_index(void)
 	return EFI_SUCCESS;
 }
 
+EFI_STATUS read_tpm2_device_state(UINT8 *state)
+{
+	EFI_STATUS ret;
+	UINT16 data_size = sizeof(UINT8);
+
+	ret = tpm2_read_nvindex(NV_INDEX_BOOTLOADER, &data_size, (BYTE *)state,
+			offsetof(tpm2_bootloader_t, lock_state));
+	if (EFI_ERROR(ret)) {
+		efi_perror(ret, L"Read device state from TPM failed");
+		goto out;
+	}
+
+	if (data_size != sizeof(UINT8)) {
+		error(L"Read device state from TPM, but data size is wrong: %d", data_size);
+		ret = EFI_COMPROMISED_DATA;
+		goto out;
```

Comment:
There is no point of having a label `out` if you can just return instead. I believe the same comment applies everywhere in this source file.